Client/Frontend/Browser/Playlist/Controller/PlaylistViewController.swift

```diff
@@ -0,0 +1,831 @@
+    }
+    
+    private func updateLayoutForOrientationChange() {
+        if playerView.isFullscreen {
+            splitController.preferredDisplayMode = .secondaryOnly
+        } else {
+            if UIDevice.current.orientation.isLandscape {
+                splitController.preferredDisplayMode = .secondaryOnly
+            } else {
+                splitController.preferredDisplayMode = .primaryOverlay
+            }
+        }
+    }
+    
+    private func updatePlayerUI() {
+        // Update play/pause button
+        if isPlaying {
+            playerView.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_pause"), for: .normal)
+        } else {
+            playerView.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_play"), for: .normal)
+        }
+        
+        // Update play-backrate button
+        let playbackRate = player.rate
+        let button = playerView.controlsView.playbackRateButton
+        
+        if playbackRate <= 1.0 {
+            button.setTitle("1x", for: .normal)
+        } else if playbackRate == 1.5 {
+            button.setTitle("1.5x", for: .normal)
+        } else {
+            button.setTitle("2x", for: .normal)
+        }
+        
+        // Update repeatMode button
+        switch repeatMode {
+        case .none:
+            playerView.controlsView.repeatButton.setImage(#imageLiteral(resourceName: "playlist_repeat"), for: .normal)
+        case .repeatOne:
+            playerView.controlsView.repeatButton.setImage(#imageLiteral(resourceName: "playlist_repeat_one"), for: .normal)
+        case .repeatAll:
+            playerView.controlsView.repeatButton.setImage(#imageLiteral(resourceName: "playlist_repeat_all"), for: .normal)
+        }
+        
+        if let item = PlaylistCarplayManager.shared.currentPlaylistItem {
+            playerView.setVideoInfo(videoDomain: item.pageSrc, videoTitle: item.pageTitle)
+        } else {
+            playerView.resetVideoInfo()
+        }
+    }
+    
+    private func observePlayerStates() {
+        player.publisher(for: .play).sink { [weak self] _ in
+            self?.playerView.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_pause"), for: .normal)
+        }.store(in: &playerStateObservers)
+        
+        player.publisher(for: .pause).sink { [weak self] _ in
+            self?.playerView.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_play"), for: .normal)
+        }.store(in: &playerStateObservers)
+        
+        player.publisher(for: .stop).sink { [weak self] _ in
+            self?.playerView.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_play"), for: .normal)
+        }.store(in: &playerStateObservers)
+        
+        player.publisher(for: .changePlaybackRate).sink { [weak self] _ in
+            guard let self = self else { return }
+            
+            let playbackRate = self.player.rate
```

Comment:
I wonder if you could get playback rate directly from the publisher instead of having to access it from `player` ?